File: frontend/src/api/order.ts
import { http } from '@/utils/request'
import type { PageResult, PageParams } from '@/types'

/**
 * 订单信息
 */
export interface OrderInfo {
  id: string
  orderNo: string  // 订单编号
  tableId: number
  tableName: string
  startTime: number
  endTime: number | null
  duration: number
  pauseDuration: number
  presetDuration: number | null
  channel: string  // 订餐渠道
  status: 'active' | 'completed'
  amount: number
  originalAmount?: number  // 原价（折扣前）
  discountAmount?: number  // 折扣金额
  operatorName: string
  paidAt: number | null
  createdAt: number
  memberId?: number | null  // 会员ID
  memberName?: string  // 会员姓名（用于显示）
  memberLevelName?: string  // 会员等级名称
  memberDiscountRate?: number  // 会员折扣率
  paymentMethod?: string  // 支付方式
  balanceAmount?: number  // 余额支付金额
  otherPaymentAmount?: number  // 其他方式支付金额
}

/**
 * 订单查询参数
 */
export interface OrderQueryParams extends PageParams {
  status?: string
  startTime?: number
  endTime?: number
  tableId?: number
  keyword?: string
}

/**
 * 时间线记录
 */
export interface TimeLineItem {
  time: number
  action: string
  description: string
  operator: string
}

/**
 * 订单详情
 */
export interface OrderDetail {
  id: string
  tableId: number
  tableName: string
  startTime: number
  endTime: number | null
  paidAt: number | null
  duration: number
  pauseDuration: number
  actualDuration: number
  presetDuration: number | null
  channel?: string  // 订单渠道
  status: 'active' | 'completed'
  amount: number
  originalAmount?: number  // 原价（折扣前）
  discountAmount?: number  // 折扣金额
  normalAmount: number
  overtimeAmount: number
  operatorId: string
  operatorName: string
  createdAt: number
  updatedAt: number
  memberId?: number | null  // 会员ID
  memberName?: string  // 会员姓名
  memberLevelName?: string  // 会员等级名称
  memberDiscountRate?: number  // 会员折扣率
  paymentMethod?: string  // 支付方式
  balanceAmount?: number  // 余额支付金额
  otherPaymentAmount?: number  // 其他方式支付金额
  timeLine: TimeLineItem[]
}

/**
 * 获取当前订单（分页）
 */
export const getActiveOrders = (params?: PageParams) => {
  return http.get<PageResult<OrderInfo>>('/orders/active', { params })
}

/**
 * 获取历史订单
 */
export const getHistoryOrders = (params: OrderQueryParams) => {
  return http.get<PageResult<OrderInfo>>('/orders/history', { params })
}

/**
 * 获取订单详情
 */
export const getOrderDetail = (id: string) => {
  return http.get<OrderDetail>(`/orders/${id}`)
}

/**
 * 导出订单
 */
export const exportOrders = (params: OrderQueryParams) => {
  return http.get<Blob>('/orders/export', {
    params,
    responseType: 'blob'
  })
}

File: backend/pdsystem-entity/src/main/java/com/pindou/timer/entity/Order.java
package com.pindou.timer.entity;

import com.baomidou.mybatisplus.annotation.*;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;

/**
 * 订单实体类
 *
 * @author pindou
 * @since 1.0.0
 */
@Data
@TableName("biz_order")
@Schema(description = "订单")
public class Order implements Serializable {

    private static final long serialVersionUID = 1L;

    @Schema(description = "订单ID（UUID）")
    @TableId(value = "id", type = IdType.INPUT)
    @TableField(typeHandler = org.apache.ibatis.type.StringTypeHandler.class)
    private String id;

    @Schema(description = "父订单ID（NULL表示为父订单/独立订单）")
    @TableField("parent_order_id")
    private String parentId;

    @Schema(description = "桌台ID")
    @TableField("table_id")
    private Integer tableId;

    @Schema(description = "桌台名称")
    @TableField("table_name")
    private String tableName;

    @Schema(description = "开始时间（毫秒时间戳）")
    @TableField("start_time")
    private Long startTime;

    @Schema(description = "结束时间（毫秒时间戳）")
    @TableField("end_time")
    private Long endTime;

    @Schema(description = "总时长（秒）")
    @TableField("duration")
    private Integer duration;

    @Schema(description = "暂停总时长（秒）")
    @TableField("pause_duration")
    private Integer pauseDuration;

    @Schema(description = "预设时长（秒）")
    @TableField("preset_duration")
    private Integer presetDuration;

    @Schema(description = "订餐渠道")
    @TableField("channel")
    private String channel;

    @Schema(description = "状态：active-进行中 completed-已完成")
    @TableField("status")
    private String status;

    @Schema(description = "总金额")
    @TableField("amount")
    private BigDecimal amount;

    @Schema(description = "金额明细（JSON）")
    @TableField("amount_detail")
    private String amountDetail;

    @Schema(description = "操作员ID")
    @TableField("operator_id")
    private String operatorId;

    @Schema(description = "操作员姓名")
    @TableField("operator_name")
    private String operatorName;

    @Schema(description = "会员ID")
    @TableField("member_id")
    private Long memberId;

    @Schema(description = "活动折扣ID")
    @TableField("discount_id")
    private String discountId;

    @Schema(description = "活动折扣名称")
    @TableField("discount_name")
    private String discountName;

    @Schema(description = "活动折扣率（0.9表示9折）")
    @TableField("discount_rate")
    private BigDecimal discountRate;

    @Schema(description = "活动折扣金额")
    @TableField("discount_amount")
    private BigDecimal discountAmount;

    @Schema(description = "会员折扣金额")
    @TableField("member_discount")
    private BigDecimal memberDiscount;

    @Schema(description = "原价（折扣前）")
    @TableField("original_amount")
    private BigDecimal originalAmount;

    @Schema(description = "支付时间（毫秒时间戳）")
    @TableField("paid_at")
    private Long paidAt;

    @Schema(description = "支付方式：offline-线下, online-线上, balance-余额, combined-组合")
    @TableField("payment_method")
    private String paymentMethod;

    @Schema(description = "余额支付金额")
    @TableField("balance_amount")
    private BigDecimal balanceAmount;

    @Schema(description = "其他方式支付金额")
    @TableField("other_payment_amount")
    private BigDecimal otherPaymentAmount;

    @Schema(description = "创建时间（毫秒时间戳）")
    @TableField(value = "created_at", fill = FieldFill.INSERT)
    private Long createdAt;

    @Schema(description = "更新时间（毫秒时间戳）")
    @TableField(value = "updated_at", fill = FieldFill.INSERT_UPDATE)
    private Long updatedAt;
}
